test/cypress/tests/arrow-navigation.spec.ts

```diff
@@ -0,0 +1,444 @@
+
+          expect(range.startOffset).to.be.closeTo('Second paragraph'.length, 3);
+          expect(range.endOffset).to.be.closeTo('Second paragraph'.length, 3);
+          expect(range.startContainer.parentElement.closest('.ce-block').textContent).to.eq('Some long long long text');
+        });
+    });
+
+    it('should navigate to previous block from empty block by arrow up', () => {
+      cy.get('[data-cy=editorjs]')
+        .find('.ce-block')
+        .eq(1)
+        .type('{movetoend}{enter}')
+        .next()
+        .type('{uparrow}');
+
+      cy.window()
+        .then(win => {
+          const selection = win.getSelection();
+          const range = selection.getRangeAt(0);
+
+          expect(range.startOffset).to.eq(0);
+          expect(range.endOffset).to.eq(0);
+          expect(range.startContainer.parentElement.closest('.ce-block').textContent).to.eq('Second paragraph');
+        });
+    });
+
+    it('should navigate from inline fragment to previous block by arrow down', () => {
+      cy.get('[data-cy=editorjs]')
+        .find('.ce-block')
+        .first()
+        .type('Some text')
+        .next()
+        .type('{movetoend}{uparrow}');
+
+      cy.window()
+        .then(win => {
+          const selection = win.getSelection();
+          const range = selection.getRangeAt(0);
+
+          expect(range.startOffset).to.closeTo('Some text'.length, 1);
+          expect(range.endOffset).to.closeTo('Some text'.length, 1);
+          expect(range.startContainer.parentElement.closest('.ce-block').textContent).to.eq('Some text');
+        });
+    });
+
+    it('should navigate to empty block by up down', () => {
+      cy.get('[data-cy=editorjs]')
+        .find('.ce-block')
+        .eq(1)
+        .type('{movetoend}{uparrow}');
+
+      cy.window()
+        .then(win => {
+          const selection = win.getSelection();
+          const range = selection.getRangeAt(0);
+
+          expect(range.startOffset).to.eq(0);
+          expect(range.endOffset).to.eq(0);
+          expect(range.startContainer.parentElement.closest('.ce-block').textContent).to.eq('');
+        });
+    });
+
+    it('should save caret horizontal offset on navigation by arrow down', () => {
+      cy.get('[data-cy=editorjs]')
+        .find('.ce-block')
+        .eq(1)
+        .type('{movetoend}{enter}');
+
+      cy.get('[data-cy=editorjs]')
+        .find('.ce-block')
+        .last()
+        .type('{movetostart}' + '{rightarrow}'.repeat(5))
+        .type('{uparrow}')
+        .prev()
+        .type('{uparrow}');
+
+      cy.window()
+        .then(win => {
+          const selection = win.getSelection();
+          const range = selection.getRangeAt(0);
+
+          expect(range.startOffset).to.be.closeTo(5, 1);
+          expect(range.endOffset).to.be.closeTo(5, 1);
+          expect(range.startContainer.parentElement.closest('.ce-block').textContent).to.eq('Second paragraph');
+        });
+    });
+  });
```

Comment:
there should be few more cases:

1. Arrow down (and right at the end) at the last **non-default** block should create the new block below and set caret to it
2. Arrow down (and right at the end) at the last **default** block should not create the new block below